website/pages/api-docs/connect/intentions.mdx

```diff
@@ -12,7 +12,134 @@ description: |-
 The `/connect/intentions` endpoint provide tools for managing
 [intentions](/docs/connect/intentions).
 
-## Create Intention
+-> **1.9.0 and later:** Reading and writing intentions has been
+migrated to the
+[`service-intentions`](/docs/agent/config-entries/service-intentions)
+config entry kind.
+
+## Upsert Intention by Name <sup>Beta</sup>
+
+-> **1.9.0+:** This API is available in Consul versions 1.9.0 and later.
+
+This endpoint creates a new intention and returns `true` if it was created
+successfully.
+
+The name and destination pair must be unique. If another intention matches the
+name and destination, the creation will replace the previous intention.
+
+~> The intentions created by this endpoint will not be assigned the following
+fields: `ID`, `CreatedAt`, `UpdatedAt`. Additionally, the `Meta` field cannot
+be persisted using this endpoint and will require editing the enclosing
+`service-intentions` config entry directly for the destination.
+
+| Method | Path                        | Produces           |
+| ------ | --------------------------- | ------------------ |
+| `PUT`  | `/connect/intentions/exact` | `application/json` |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/features/blocking),
+[consistency modes](/api/features/consistency),
+[agent caching](/api/features/caching), and
+[required ACLs](/api#authentication).
+
+| Blocking Queries | Consistency Modes | Agent Caching | ACL Required                   |
+| ---------------- | ----------------- | ------------- | ------------------------------ |
+| `NO`             | `none`            | `none`        | `intentions:write`<sup>1</sup> |
+
+<p>
+  <sup>1</sup> Intention ACL rules are specified as part of a `service` rule.
+  See{' '}
+  <a href="/docs/connect/intentions#intention-management-permissions">
+    Intention Management Permissions
+  </a>{' '}
+  for more details.
+</p>
+
+### URL Parameters
+
+- `source` `(string: <required>)` - Specifies the source service. This
+  is specified as part of the URL.
+  This can take [several forms](/docs/commands/intention#source-and-destination-naming).
+
+- `destination` `(string: <required>)` - Specifies the destination service. This
+  is specified as part of the URL.
+  This can take [several forms](/docs/commands/intention#source-and-destination-naming).
+
+- `ns` `(string: "")` <EnterpriseAlert inline /> - Specifies the default
+  namespace to use when `source` or `destination` parameters lack namespaces.
+  If not provided at all, the default namespace will be inherited from the
+  request's ACL token or will default to the `default` namespace.
+  This value may be provided by either the `ns` URL query parameter or in the
+  `X-Consul-Namespace` header.
+  Added in Consul 1.9.0.
+
+### PUT Body Parameters
+
+- `SourceType` `(string: <required>)` - The type for the `SourceName` value.
```

Comment:
If there's only one option and it's required, do we apply it as a default? In other words, is it okay if the user neglects this?

Reply:
Good thing to note. This section was unfortunately duplicated from the old endpoint for creating legacy intentions. I can update both to reflect this.